apps/src/lab2/Lab2MetricsReporter.ts

```diff
@@ -0,0 +1,84 @@
+import MetricsReporter from '../lib/metrics/MetricsReporter';
+import {MetricDimension} from '../lib/metrics/types';
+
+interface ReportingProperties {
+  channelId?: string;
+  projectType?: string;
+  currentLevelId?: string;
+  scriptId?: number;
+}
+
+/**
+ * Singleton metrics reporter for Lab2 labs.
+ */
+class Lab2MetricsReporter {
+  private commonProperties: ReportingProperties = {};
+
+  public updateProperties(properties: ReportingProperties) {
```

Comment:
Maybe worth a comment here, and for other functions that newcomers will end up using?  We might want to explain concepts such as `ReportingProperties`, too.

Reply:
Thanks for the updates.  Not sure if more coming, but I'm still not sure what dimensions are in this file :)

Reply:
Ah fair :) these are modeled after [Cloudwatch's metric dimensions](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/cloudwatch_concepts.html#Dimension), which are key-value pairs for adding various identification characteristics to metrics. It would probably make sense to add more documentation in the lib/metrics directory directly, which I can do as a follow up.

Reply:
Sounds good!